src/ui/public/vislib/visualizations/point_series/area_chart.js

```diff
@@ -29,17 +29,17 @@ export default function AreaChartFactory(Private) {
    * chart
    */
   class AreaChart extends PointSeriesChart {
-    constructor(handler, chartEl, chartData, chartConfig) {
+    constructor(handler, chartEl, chartData, seriesConfigArgs) {
       super(handler, chartEl, chartData);
 
-      this.isOverlapping = (handler.visConfig.get('mode') === 'overlap');
-
+      this.seriesConfig = _.defaults(seriesConfigArgs || {}, defaults);
```

Comment:
I think this separation is good, but I also think that `seriesConfig` should have an api similar to the rest of the `XYZConfig` classes.